src/types.ts

```diff
@@ -18,6 +34,68 @@ export enum NFTStandard {
   E1155 = 1,
 }
 
+// A RenftContractType represents a unique "style" of rental functionality:
+//
+// + Azrael: A collateralized rental; an up-front fee is required to prevent a renter from leaving a lender empty-handed.
+// + Sylvester: A non-collateralized rental; have an escrow contract programmatically safeguard the safety of your asset.
+// + Whoopi A non-collateralized rental which is cycle-driven, and includes revenue-share for counterparties.
```

Comment:
`Whoopi: A non [...]`

Reply:
cycle-driven is a convention that does not touch the contract itself. Let's use this description:

`Whoopi: A collateral-free rental; very similar to Sylvester, with an addition of whitelisting and reward share features that facilitate guild-scholar management. Push approach to pay function.`